mw-webapp/src/logic/wayPage/WayPage.tsx

```diff
@@ -165,14 +221,20 @@ export const WayPage = (props: WayPageProps) => {
             isWayInFavorites && setUser &&
             <Button
               value={"Remove from favorite"}
-              onClick={() => deleteFavoriteWayUuid(user, way.uuid, setUser)}
+              onClick={() => {
+                deleteFavoriteWayUuidFromUser(user, way.uuid, setUser);
+                deleteFavoriteForUserUuidsFromWay(way, user.uuid, setWay);
+              }}
             />
           }
           {
             !isWayInFavorites && user && setUser &&
             <Button
               value={"Add to favorite"}
-              onClick={() => addFavoriteWayUuid(user, way.uuid, setUser)}
+              onClick={() => {
+                addFavoriteWayUuidToUser(user, way.uuid, setUser);
+                addFavoriteForUserUuidsToWay(way, user.uuid, setWay);
+              }}
```

Comment:
Judging by what those functions do, you should merge it together, because no one shouldnt be able to just delete or add only WayUuid to user for example

Reply:
Done.